pkg/catalog/interrogate.go

```diff
@@ -193,21 +191,22 @@ func AppSetupPackageUrl(
 		return "", err
 	}
 
-	var appConfigUrl string
+	var appConfigURL = ""
 
 	// Check to see if there is a role-specific setup package.
 	for _, nodeRole := range appCR.Spec.NodeRoles {
 		if nodeRole.ID == role {
-			appConfigUrl = nodeRole.SetupPackage.ImportUrl
+			setupPackage := nodeRole.SetupPackage
+
+			if (setupPackage.IsSet == true) && (setupPackage.IsNull == false) {
+				appConfigURL = setupPackage.PackageURL.PackageURL
+			}
+
 			break
 		}
 	}
-	// If role-specific package is not present, use the main one.
-	if appConfigUrl == "" {
-		appConfigUrl = cr.AppSpec.Spec.SetupPackage.ImportUrl
-	}
 
-	return appConfigUrl, nil
+	return appConfigURL, nil
 }
```

Comment:
And the situation with this loop is different since it's possible to find the role but still have no setup package. So this loop should be ok as is.

Reply:
By this time, yes. I guess you are wondering if we can remove that check? Let me see.

Reply:
Cleaned it up a bit.